### PR TITLE
RNMobile: Fix broken documentation link

### PR DESCRIPTION
**Pull request by guarani**
## Description

@mchowning reported that the new **React Native Integration Test Guide**'s link was broken in the Block Editor handbook (which is generated from this repo's `docs/` folder): see https://developer.wordpress.org/block-editor/contributors/code/#contributor-resources.

When adding this new doc in https://github.com/WordPress/gutenberg/pull/33833, I'd overlooked adding it to the `docs/manifest.json` file.

## How has this been tested?
Verify that this change follows the guideline in as outlined in [here](https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/documentation#create-a-new-document).

After merging, expect the fix to be reflected in the Block Editor handbook ([after 15 minutes](https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/documentation#block-editor-handbook-process)).

## Types of changes
- Doc fixes

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: docs/contributors/code/testing-overview.md [hunk 1]
```diff
@@ -380,11 +380,11 @@ To locally run the tests in debug mode, follow these steps:
 
 ### Native mobile end-to-end tests
 
-Contributors to Gutenberg will note that PRs include continuous integration E2E tests running the native mobile E2E tests on Android and iOS. For troubleshooting failed tests, check our guide on [native mobile tests in continious integration](docs/contributors/native-mobile.md#native-mobile-e2e-tests-in-continuous-integration). More information on running these tests locally can be found in the [relevant directory README.md](https://github.com/WordPress/gutenberg/tree/HEAD/packages/react-native-editor/__device-tests__).
+Contributors to Gutenberg will note that PRs include continuous integration E2E tests running the native mobile E2E tests on Android and iOS. For troubleshooting failed tests, check our guide on [native mobile tests in continious integration](/docs/contributors/native-mobile.md#native-mobile-e2e-tests-in-continuous-integration). More information on running these tests locally can be found in the [relevant directory README.md](https://github.com/WordPress/gutenberg/tree/HEAD/packages/react-native-editor/__device-tests__).
```
Comment (guarani) [inline docs/contributors/code/testing-overview.md]: The missing leading slash wasn't necessarily causing issues, but based on the guidelines [here](https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/documentation#using-links) and the other links I see, always using a leading slash is the convention.

File: docs/contributors/code/testing-overview.md [hunk 2]
```diff
@@ -380,11 +380,11 @@ To locally run the tests in debug mode, follow these steps:
...
 
 ### Native mobile integration tests
 
-There is an ongoing effort to add integration tests to the native mobile project using the [`react-native-testing-library`](https://testing-library.com/docs/react-native-testing-library/intro/) library. A guide to writing integration tests can be found [here](native-mobile-integration-test-guide.md).
+There is an ongoing effort to add integration tests to the native mobile project using the [`react-native-testing-library`](https://testing-library.com/docs/react-native-testing-library/intro/) library. A guide to writing integration tests can be found [here](/docs/contributors/code/native-mobile-integration-test-guide.md).
```
Comment (guarani) [inline docs/contributors/code/testing-overview.md:387]: Same as https://github.com/WordPress/gutenberg/pull/34187/files#r692558648.

File: docs/toc.json [hunk 3]
```diff
@@ -178,11 +178,13 @@
...
-				{ "docs/how-to-guides/widgets/opting-out.md": [] },
-				{ "docs/how-to-guides/widgets/legacy-widget-block.md": [] }
-			] }
+			{
+				"docs/how-to-guides/widgets/README.md": [
+					{ "docs/how-to-guides/widgets/overview.md": [] },
+					{ "docs/how-to-guides/widgets/opting-out.md": [] },
+					{ "docs/how-to-guides/widgets/legacy-widget-block.md": [] }
+				]
+			}
```
Comment (guarani) [inline docs/toc.json:187]: This change was a result of simply saving the file in Visual Studio Code. Based on the rest of the file, I think this was simply badly formatted and it makes sense to correct its format here.

File: docs/contributors/code/testing-overview.md (hunk 1 above)
Comment (fluiddot) [reply docs/contributors/code/testing-overview.md]: Looks like the path of the document is wrong, the file `native-mobile.md` is located in [`/docs/contributors/code/native-mobile.md`](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/native-mobile.md) (it's missing the `code` folder).

Comment (fluiddot) [reply docs/contributors/code/testing-overview.md]: I also found a typo: `continious` => `continuous`.

File: docs/toc.json (hunk 3 above)
Comment (fluiddot) [reply docs/toc.json:187]: I tried formatting the file locally and I had the same result 👍 .

**Review by fluiddot — COMMENTED** [review]
I added a couple of comments that would be great to be addressed, apart from that I checked the guideline and confirm it complies with it 🎊 :

- [x] Create a Markdown file in the docs folder, use lower-case, no spaces, if needed a dash separator, and .md extension.
- [x] Add content, all documents require one and only H1 tag, using markdown notation.
- [x] Add item to the toc.json hierarchy, see existing entries for format.
- [x] Run npm run docs:build to update manifest.json.
- [x] Commit manifest.json with other files updated.

File: docs/contributors/code/testing-overview.md (hunk 1 above)
Comment (guarani) [reply docs/contributors/code/testing-overview.md]: Thanks for catching those two!

Comment (guarani) [reply docs/contributors/code/testing-overview.md]: Fixed now.

**Comment by guarani** [comment]
Thanks for the review, @fluiddot!

---

👋 @mchowning, would you be able to give this a review please? @fluiddot will be away, so I needed a new reviewer :)

File: docs/contributors/code/testing-overview.md [hunk 4]
```diff
@@ -380,11 +380,11 @@ To locally run the tests in debug mode, follow these steps:
 
 ### Native mobile end-to-end tests
 
-Contributors to Gutenberg will note that PRs include continuous integration E2E tests running the native mobile E2E tests on Android and iOS. For troubleshooting failed tests, check our guide on [native mobile tests in continious integration](docs/contributors/native-mobile.md#native-mobile-e2e-tests-in-continuous-integration). More information on running these tests locally can be found in the [relevant directory README.md](https://github.com/WordPress/gutenberg/tree/HEAD/packages/react-native-editor/__device-tests__).
+Contributors to Gutenberg will note that PRs include continuous integration E2E tests running the native mobile E2E tests on Android and iOS. For troubleshooting failed tests, check our guide on [native mobile tests in continuous integration](/docs/contributors/code/native-mobile.md#native-mobile-e2e-tests-in-continuous-integration). More information on running these tests locally can be found in the [here](/packages/react-native-editor/__device-tests__/README.md).
```
Comment (carolinan) [inline docs/contributors/code/testing-overview.md]: I believe this needs updating, it says: `found in the here`?

Comment (gziolo) [inline docs/contributors/code/testing-overview.md]:
```suggestion
Contributors to Gutenberg will note that PRs include continuous integration E2E tests running the native mobile E2E tests on Android and iOS. For troubleshooting failed tests, check our guide on [native mobile tests in continuous integration](/docs/contributors/code/native-mobile.md#native-mobile-e2e-tests-in-continuous-integration). More information on running these tests locally can be found in [here](/packages/react-native-editor/__device-tests__/README.md).
```

**Review by gziolo — APPROVED** [review]
👍🏻

**Review by mchowning — APPROVED** [review]

File: docs/contributors/code/testing-overview.md (hunk 4 above)
Comment (guarani) [reply docs/contributors/code/testing-overview.md]: Thanks for catching that!

Comment (guarani) [reply docs/contributors/code/testing-overview.md]: Thanks for catching that, fixed in 1c3300cd6a172e31827eec3408ffe5acfce79f07.

**Comment by guarani** [comment]
FYI: I checked the "React Native Integration Test Guide" link under https://developer.wordpress.org/block-editor/contributors/code/#contributor-resources and it is now working 👍

**Pull request merged**